server/service/hosts.go

```diff
@@ -939,8 +940,8 @@ func (svc *Service) getHostDetails(ctx context.Context, host *fleet.Host, opts f
 
 				// determine disk encryption and action required here based on profiles and
 				// raw decryptable key status.
-				host.MDM.DetermineMacOSDiskEncryptionStatus(profs, mobileconfig.FleetFileVaultPayloadIdentifier)
-				host.MDM.OSSettings.DiskEncryption.Status = host.MDM.MacOSSettings.DiskEncryption
+				hde := host.MDM.DetermineHostMDMDiskEncryptionMacOS(profs, mobileconfig.FleetFileVaultPayloadIdentifier)
```

Comment:
the only problem I see with this is that it was clear previously that this function was modifying the internal status of the struct.

Now it additionally returns a value, which makes things a bit confusing. I'm OK to move forward as-is since this is a hotfix, but wanted to point that out.

Reply:
Yeah, I share your concerns about that too and I made a point to call it out in the doc comments for the method. I'm not sure the best way to handle it other than creating function function that also iterates over the profiles and returns the new struct.

Reply:
I guess I'm missing the reason why it can't set `host.MDM.OSSettings.DiskEncryption` directly in the method call? It is part of the `MDMHostData` struct on which the method lives.

Reply:
Good point, I'll improve this in a follow up PR that also adds some integration tests.